TodoWithControllersAuthJWT/Program.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Tokens;
+using TodoWithControllersAuthJWT;
+
+namespace Todos
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            builder.Services.AddDbContext<TodoDbContext>(options => options.UseInMemoryDatabase("Todos"));
+            builder.Services.AddSingleton<IAuthService, AuthService>();
+            builder.Services.AddAuthorization(options =>
+            {
+                options.AddPolicy("admin", policy => policy.RequireClaim("can_delete", "true"));
+                options.AddPolicy("user", policy => policy.RequireClaim("can_view", "true"));
+            });
+            var authService = builder.Services.BuildServiceProvider().GetService<IAuthService>();
```

Comment:
Noooo, this is a nono, though shall not call `BuildServiceProvider` 😄. Looks like we need the key in 2 places. Maybe it's fine to leave it here and reference it from the controller and here. 

Let me think about this a bit.

Reply:
OK so I messed up originally. The Issuer and Key should not be on the `IAuthService`. That's specially about user/password/claims management. The key and issuer usually come from configuration. For the sake of keeping things simple, we can specify those in Program or another static class and use them from both Program.cs and AuthController.cs